### PR TITLE
Rebased version of the elastic PR

### PR DESCRIPTION
**Pull request by jacksonh**
- Add elastic to our docker compose
- add AND/OR/NOT search operations
- add elastic and create article in elastic
- change error code when elastic throws error
- add search pages in elastic
- add search by labels
- Add elastic to GitHub Action
- Update elastic version
- Fix port for elastic
- add url in search query
- Set elastic features when running tests
- add debug logs
- Use localhost instead of service hostname
- refresh elastic after create/update
- update search labels query
- add typescript support
- search pages in elastic
- fix search queries
- use elastic for saving page
- fix test failure
- update getArticle api to use elastic
- use generic get page function
- add elastic migration python script
- fix bulk helper param
- save elastic page id in article_saving_request instead of postgres article_id
- fix page archiving and deleting
- add tests for deleteArticle
- remove custom date type in elastic mappings which not exist in older version of elastic
- fix timestamp format issue
- add tests for save reading progress
- add tests for save file
- optimize search results
- add alias to index
- update migration script to receive env var as params
- Add failing test to validate we don't decrease reading progress
- Rename readingProgress to readingProgressPercent
- Linting
- Add failing test for creating highlights w/elastic
- add tests code coverage threshold
- update nyc config
- include more files in test coverage
- change alias name
- update updateContent to update pages in elastic
- remove debug log
- fix createhighlight test
- search pages by alias in elastic
- update set labels and delete labels in elastic
- migration script enumeration
- make BULK_SIZE an env var
- fix pdf search indexing
- debug github action exit issue
- call pubsub when create/update/delete page in elastic
- fix json parsing bug and reduce reading data from file
- replace a depreciated pubsub api call
- debug github action exit issue
- debug github action exit issue
- add handler to upload elastic page data to GCS
- fix tests
- Use http_auth instead of basic_auth
- add index creation and existing postgres tables update in migration script
- fix a typo to connect to elastic
- rename readingProgress to readingProgressPercent
- migrate elastic_page_id in highlights and article_saving_request tables
- update migration script to include number of updated rows
- update db migration query
- read index mappings from file
- fix upload pages to gcs
- fix tests failure due to pageContext
- fix upload file id not exist error
- Handle savedAt & isArchived attributes w/out quering elastic
- fix content-type mismatching
- revert pageId to linkId because frontend was not deployed yet
- fix newsletters and attachment not saved in elastic
- put linkId in article for setting labels
- exclude orginalHtml and content in the result of searching to improve performace
- remove score sorting
- do not refresh immediately to reduce searching and indexing time
- do not replace the backup data in gcs
- fix no article id defined in articleSavingRequest
- reduce home feed pagination size to 10
- stop revalidating first page
- do not use a separate api to fetch reading progress

### CODE REVIEW

**Comment by jacksonh** [comment]
@sywhb just created this rebased version to deploy to demo.

**Review by sywhb — APPROVED** [review]

**Comment by sywhb** [comment]
> @sywhb just created this rebased version to deploy to demo.

Should I push new commit to this branch if there is any? @jacksonh

**Comment by sywhb** [comment]
> > @sywhb just created this rebased version to deploy to demo.
> 
> Should I push new commit to this branch if there is any? @jacksonh

Ignore this

File: packages/api/elastic_migrate.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+UPDATE_ARTICLE_SAVING_REQUEST_SQL = f'''
+UPDATE omnivore.article_saving_request
+    SET elastic_page_id = article_id
+    WHERE elastic_page_id is NULL
+        AND article_id is NOT NULL
+        AND updated_at > '{UPDATE_TIME}';
+'''
+
+UPDATE_HIGHLIGHT_SQL = f'''
```
Comment (jacksonh) [inline packages/api/elastic_migrate.py:58]: Do we need to do this update on other tables I see these ones in the migration (but then see the columns are removed later): 

```
ALTER TABLE omnivore.article_saving_request ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.reaction ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.highlight ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.reminders ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.abuse_report ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.content_display_report ADD COLUMN elastic_page_id varchar(36);
ALTER TABLE omnivore.link_share_info ADD COLUMN elastic_page_id varchar(36);
```

Comment (sywhb) [reply packages/api/elastic_migrate.py:58]: We probably do not need to do because these tables are not being used for now @jacksonh

**Pull request merged**